src/ocrmypdf/_metadata.py

```diff
@@ -52,7 +52,8 @@ def from_document_info(key):
 
     pdfmark['/Creator'] = f'{PROGRAM_NAME} {OCRMYPF_VERSION} / {creator_tag}'
     pdfmark['/Producer'] = f'pikepdf {PIKEPDF_VERSION}'
-    pdfmark['/ModDate'] = encode_pdf_date(datetime.now(timezone.utc))
+    if not options.deterministic_output:
+        pdfmark['/ModDate'] = encode_pdf_date(datetime.now(timezone.utc))
```

Comment:
In addition to leaving ModDate undisturbed, we should use the environment variables used to control deterministic builds when compiling.

https://blog.conan.io/2019/09/02/Deterministic-builds-with-C-C++.html

In brief, if set ModDate to SOURCE_DATE_EPOCH which would be an environment variable holding a Unix epoch timestamp (seconds since 1970). If ZERO_AR_DATE is 1, then set ModDate to 0 (1970). That allows the user to control the timestamp.